### PR TITLE
feat(t04) : search jobs by title

### PR DESCRIPTION
**Pull request by HoangTrieu2803**

### CODE REVIEW

File: src/containers/Home/Jobs/Pagination/index.tsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+import { RootState } from "../../../Redux/store/store";
 
-export default function Pagination(props: { jobsPerPage: number, totalJobs: number, paginate: any, currentPage: number }) {
+export default function Pagination(props: { jobsPerPage: number, totalJobs: number, paginate: any, currentPage: number,totalSearchJob: number }) {
 
-    const { jobsPerPage, totalJobs, paginate, currentPage } = props
+    const { jobsPerPage, totalJobs, paginate, currentPage, totalSearchJob } = props
     const [pageNumber, setPageNumber] = useState<Number[]>([]);
-    const totalPage = Math.ceil(totalJobs / jobsPerPage);
+    const search = useSelector((state:RootState)=> state.search)
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/Pagination/index.tsx:9]: should `{ search } = useSelector(...)` for shorten

File: src/containers/Home/Jobs/Pagination/index.tsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+    let totalPage = 0;
+    search.search == "" ? totalPage = Math.ceil(totalJobs / jobsPerPage) : totalPage = Math.ceil(totalSearchJob / jobsPerPage)
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/Pagination/index.tsx:11]: use strict compare and try to do this logic into useEffect

File: src/containers/Home/Jobs/index.tsx [hunk 3]
```diff
@@ -12,6 +12,8 @@ import { faArrowLeft, faArrowRight } from '@fortawesome/free-solid-svg-icons';
 export default function Jobs() {
     const dispatch = useDispatch();
     const { data: jobs, loading } = useSelector((state: RootState) => state.jobs)
+    const search = useSelector((state:RootState)=> state.search)
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:15]: should `{ search } = useSelector(...)`

File: src/containers/Home/Jobs/index.tsx [hunk 4]
```diff
@@ -12,6 +12,8 @@ import { faArrowLeft, faArrowRight } from '@fortawesome/free-solid-svg-icons';
...
+    console.log(search.search)
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:16]: remove `console.log`

File: src/containers/Home/Jobs/index.tsx [hunk 5]
```diff
@@ -31,18 +32,33 @@ export default function Jobs() {
     const handleClickNextPage = () => {
         setCurrentPage(currentPage + 1)
     }
-
+    useEffect(()=>{
+        setCurrentPage(1)
+    },[search])
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:37]: put hook on the top of component

File: src/containers/Home/Jobs/index.tsx [hunk 6]
```diff
@@ -31,18 +32,33 @@ export default function Jobs() {
...
+    const searchJob = () =>{
+        return jobs.filter((searchJob)=>{
+            if(search.search === ""){
+                return searchJob
+            }
+            else if(searchJob.jobName.toLocaleLowerCase().includes(search.search.toLocaleLowerCase())){
+                return searchJob
+            }
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/index.tsx:45]: just `return searchJob.jobName.toLocaleLowerCase().includes(search.toLocaleLowerCase())`

File: src/containers/Home/Search/index.tsx [hunk 7]
```diff
@@ -1,10 +1,21 @@
...
+import { useSelector,useDispatch } from 'react-redux';
 import { RootState } from '../../Redux/store/store';
 import Job from '../../model/job';
-
+import { searchAction } from "../../Redux/search/searchSlice";
 function Search() {
     const { data: jobs } = useSelector((state: RootState) => state.jobs)
+    const dispatch = useDispatch()
+    const [title , setTitle] = useState("");
+    const handleOnchange = (event : any) =>{
```
Comment (nqbinh13) [inline src/containers/Home/Search/index.tsx:11]: avoid using `any`

File: src/containers/Home/Search/index.tsx [hunk 8]
```diff
@@ -1,10 +1,21 @@
...
+        const name = event.target.value
+        setTitle(name) 
+    }
+    const handleSearch = (event : any) =>{
```
Comment (nqbinh13) [inline src/containers/Home/Search/index.tsx:15]: avoid using `any`

File: src/containers/Home/Jobs/Pagination/index.tsx (hunk 1 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/Pagination/index.tsx]: i've changed

File: src/containers/Home/Jobs/Pagination/index.tsx (hunk 2 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/Pagination/index.tsx]: i've changed

File: src/containers/Home/Jobs/index.tsx (hunk 3 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i've changed

File: src/containers/Home/Jobs/index.tsx (hunk 4 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i've removed

File: src/containers/Home/Jobs/index.tsx (hunk 5 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i've put

File: src/containers/Home/Jobs/index.tsx (hunk 6 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/index.tsx]: i've changed

File: src/containers/Home/Search/index.tsx (hunk 7 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.tsx]: i've changed

File: src/containers/Home/Search/index.tsx (hunk 8 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.tsx]: i've changed

File: src/containers/Home/Search/index.tsx [hunk 9]
```diff
@@ -1,10 +1,21 @@
...
 import Job from '../../model/job';
-
+import { searchAction } from "../../Redux/search/searchSlice";
 function Search() {
     const { data: jobs } = useSelector((state: RootState) => state.jobs)
+    const dispatch = useDispatch()
+    const [title , setTitle] = useState("");
+    const handleOnchange = (event : any) =>{
+        const name = event.target.value
+        setTitle(name)
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Search/index.tsx]: i've changed

**Pull request merged**